### PR TITLE
Implement omit function

### PR DESCRIPTION
**Pull request by shaharkazaz**

### CODE REVIEW

File: utils/__tests__/object.spec.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        describe('single key', () => {
+
+            it('should remove property from object', () => {
+                const result = omit(originalObj, 'f');
+                expect(result.f).toBeUndefined();
+            });
+
+            it('should remove nested property', () => {
+                const result = omit(originalObj, 'a.b.c');
+                expect(result.a.b.c).toBeUndefined();
```
Comment (NetanelBasal) [inline utils/__tests__/object.spec.ts]: add  expect(result.a.b).toBeDefined();

Comment (shaharkazaz) [reply utils/__tests__/object.spec.ts]: Added 👍

File: utils/src/object/index.ts [hunk 2]
```diff
@@ -87,37 +86,85 @@ export function diff(object, base) {
...
+ * @param {string | string[]} paths - The path/s for the properties to delete
  * @returns {Object}
  */
-export function omit(object: Object, paths: string | string[]): Object {
-  return _omit(object, paths);
+export function omit(object: Object, paths: string | string[]) {
+    if (!object) {
+        throw 'No object given';
+    }
+    if (typeof object !== 'object') {
```
Comment (NetanelBasal) [inline utils/src/object/index.ts]: not the best check, typeof null is object, use isObject.

Comment (shaharkazaz) [reply utils/src/object/index.ts]: replaced 👍

File: utils/src/object/index.ts [hunk 3]
```diff
@@ -87,37 +86,85 @@ export function diff(object, base) {
...
+    }
+    const invalidType = pathIsArray
+        ? (paths as any[]).some((key) => typeof key !== 'string')
+        : typeof paths !== 'string';
+    if (invalidType) {
+        throw 'Expected a string / string array but got something else';
+    }
+
+    const newObj = {...object};
+    if (pathIsArray) {
```
Comment (NetanelBasal) [inline utils/src/object/index.ts]: if you changed this to Array.isArray(paths) ts will know the type automatically.

Comment (shaharkazaz) [reply utils/src/object/index.ts]: I'm using Array.isArray(paths) several times, this is why it's in a variable, should I remove the variable?

File: utils/src/object/index.ts [hunk 4]
```diff
@@ -87,37 +86,85 @@ export function diff(object, base) {
...
+        return (paths as string[]).reduce((acc, key) => {
+            return omitPath(acc, key);
```
Comment (NetanelBasal) [inline utils/src/object/index.ts]: can be one line

Comment (shaharkazaz) [reply utils/src/object/index.ts]: 👍

File: utils/src/object/index.ts [hunk 5]
```diff
@@ -87,37 +86,85 @@ export function diff(object, base) {
...
+        if (!nested.length) {
+            const {[key]: deletedKey, ...others} = newObj;
+            return others;
+        }
+        newObj[key] = omitPath(newObj[key], nested.join("."));
+    } else {
+        console.warn(`${key} doesn't exists in given object`);
+    }
+    return newObj;
 }
```
Comment (NetanelBasal) [inline utils/src/object/index.ts]: Well done 👏

Comment (shaharkazaz) [reply utils/src/object/index.ts]: Thanks! 😁

File: utils/src/object/index.ts (hunk 3 above)
Comment (NetanelBasal) [reply utils/src/object/index.ts]: Add to array utils the following function and use it:

```
function isArray<T>(value: T[] | any): value is T[] {
    return Array.isArray(value);
}
```

Comment (NetanelBasal) [reply utils/src/object/index.ts]: now you can remove the as string[] and as string

**Pull request merged**